build.gradle.kts

```diff
@@ -27,26 +28,44 @@ val projects = listOf(
 
 configure(projects) {
     apply(plugin = "org.jetbrains.kotlin.jvm")
+    apply(plugin = "org.jetbrains.kotlin.plugin.serialization")
 
     java {
         sourceCompatibility = JavaVersion.VERSION_1_8
         targetCompatibility = JavaVersion.VERSION_1_8
     }
 
-    tasks.withType<KotlinCompile> {
-        kotlinOptions {
-            jvmTarget = JavaVersion.VERSION_1_8.toString()
+    tasks {
+        withType<KotlinCompile> {
+            kotlinOptions {
+                jvmTarget = JavaVersion.VERSION_1_8.toString()
+                freeCompilerArgs = listOf(
+                    "-Xopt-in=kotlin.RequiresOptIn",
+                    "-Xopt-in=kotlinx.coroutines.ExperimentalCoroutinesApi",
+                    "-Xopt-in=kotlinx.coroutines.InternalCoroutinesApi",
+                    "-Xopt-in=kotlinx.serialization.ExperimentalSerializationApi",
+                )
+            }
+        }
+
+        withType<LintTask> {
+            source(files("src/kotlin"))
+        }
+
+        withType<FormatTask> {
+            source(files("src/kotlin"))
         }
     }
 
+
     dependencies {
         // Kotlin
         implementation(kotlin("stdlib-jdk8"))
         implementation(kotlin("reflect"))
         testImplementation(kotlin("test-junit5"))
 
         // coroutines
-        val coroutinesVersion = "1.5.1"
+        val coroutinesVersion = "1.5.2"
```

Comment:
Isn't coroutines version locked by kotlin?

Reply:
It isnt usually, as long as you are not a major version behind you are good to update it as you like.